src/option_parser.cr

```diff
@@ -1,3 +1,30 @@
+# `OptionParser` is a class for command-line options processing. It supports:
+#
+# * Short and long modifier style options (example: `-h`, `--help`)
+# * Command style arguments (example: `crystal build`)
+# * Passing arguments to the flags (example: `-f filename.txt`)
+# * Automatic help message generation
+#
+# Run `crystal` for an example of a CLI built with `OptionParser`.
```

Comment:
It's ok, the compiler uses OptionParser. Maybe not without some additional configuration, but it's a good example of its use.